xtext/org.icyphy.linguafranca/src/lib/core/reactor_common.c

```diff
@@ -74,32 +64,23 @@ microstep_t current_microstep = 0;
 instant_t duration = -1LL;
 
 /**
- * Physical time at the start of the execution.
- */
-instant_t physical_start_time = 0LL;
-
-/**
- * Logical time at the start of execution.
- */
-interval_t start_time = 0LL;
-
-/**
- * Indicates whether or not the execution 
+ * Indicates whether or not the execution
  * has started.
  */
 bool _lf_execution_started = false;
 
 /**
  * Indicator that the execution should stop after the completion of the
- * current logical time. This can be set to true by calling the `stop()`
+ * current logical time. This can be set to true by calling the `request_stop()`
  * function in a reaction.
  */
 bool stop_requested = false;
 
 /**
- * Stop time (start_time + duration), or 0 if no timeout time has been given.
+ * Timeout time (start_time + duration), or -1 if no timeout time has been given.
+ * The microstep for timeout is always 0.
  */
-instant_t stop_time = 0LL;
+instant_t timeout_time = -1LL;
```

Comment:
Why this rename? Isn't this variable also used to enforce requested stops?

Also, why is the value `-1`, not `NEVER`?

Reply:
Requested stops are enforced using `bool stop_requested`. We do re-use the timeout mechanism in the case where the `STOP_GRANTED` message coming from the RTI carries a timestamp that is in the future as per the description in [Termination](https://github.com/icyphy/lingua-franca/wiki/Termination). But in terms of when shutdowns are scheduled, timeout indicates a microstep of 0 and `stop_requested` indicates the next immediate microstep.